lib/puppet/provider/netapp_aggregate/cmode.rb

```diff
@@ -99,12 +114,15 @@ def flush
     when :present
       Puppet.debug("Puppet::Provider::Netapp_aggregate.cmode: Ensure is present. Modifying...")
 
-      # Add disks to the existing aggregate
-      newdisks = Integer(@resource[:diskcount]) - Integer(@original_values[:diskcount])
-      if newdisks < 0
-        raise ArgumentError, "aggregate diskcount cannot be reduced without destroying and recreating the aggregate"
-      end
-      result = aggradd('aggregate', @resource[:name], 'disk-count', newdisks)
+      # apply options to the aggregate
+      setoptions()
+
+#      # Add disks to the existing aggregate
+#      newdisks = Integer(@resource[:diskcount]) - Integer(@original_values[:diskcount])
+#      if newdisks < 0
+#        raise ArgumentError, "aggregate diskcount cannot be reduced without destroying and recreating the aggregate"
+#      end
+#      result = aggradd('aggregate', @resource[:name], 'disk-count', newdisks)
```

Comment:
i commented this out because it makes no sense, there is no aggregate modify command, this tries to use the aggregate add command.